glambie/data/data_catalogue.py

```diff
@@ -310,22 +309,25 @@ def average_timeseries_in_catalogue(self, remove_trend: bool = True, add_trend_a
 
         # join all catalogues by start and end dates
         # the resulting dataframe has a set of columns with repeating prefixes
-        df = reduce(lambda left, right: left.merge(right, how="outer", on=["start_dates", "end_dates"]), catalogue_dfs)
-        df = df.sort_values(by="start_dates")
-        start_dates, end_dates = np.array(df["start_dates"]), np.array(df["end_dates"])
-        mean_changes = np.array(df[df.columns.intersection(
-            df.filter(regex=("changes*")).columns.to_list())].mean(axis=1))
+        df_merged_all = pd.concat([x.set_index(['start_dates', 'end_dates']) for x in catalogue_dfs],
```

Comment:
Minor one - leaving this simply as "df" is fine. Or at the very least, the last _all is a tautology, so df_merged or df_all would read better.

Reply:
done